Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -555,11 +568,13 @@ private extension PurchasesOrchestrator {
     }
 
     func handleFailedTransaction(_ transaction: SKPaymentTransaction) {
+        let storeTransaction = StoreTransaction(sk1Transaction: transaction)
+
         if let error = transaction.error,
-           let completion = getAndRemovePurchaseCompletedCallback(forTransaction: transaction) {
+           let completion = self.getAndRemovePurchaseCompletedCallback(forTransaction: storeTransaction) {
```

Comment:
on a separate note, this brings up an issue that @vegaro brought up to me yesterday (some of his thoughts are captured [here](https://revenuecats.atlassian.net/browse/CSDK-99)): 
StoreTransaction currently represents Storekit transactions + backend parsed transactions. 
But... they're really different. 

For starters, a `BackendParsedTransaction` would absolutely not make sense here, meaning that we're violating Liskov principle, right? 

We can discuss it further on the ticket, but I did want to bring up that it feels like our abstraction there might get problematic over time (it already has some inconsistencies in hybrids)

Reply:
That one is definitely my fault, I didn't realize that those represented 2 different things at the time, and because they had basically the same interface I assumed they could be represented by the same type :/